Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,590 @@
+""" The RucioFileCatalogClient is a class that allows to interface Dirac with Rucio
+"""
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+__RCSID__ = "$Id"
+
+import os
+import uuid
+import datetime
+from copy import deepcopy
+
+import DIRAC
+from DIRAC                                              import S_OK, S_ERROR, gLogger
+from DIRAC.Resources.Catalog.Utilities                  import checkCatalogArguments
+from DIRAC.Core.Utilities.List                          import breakListIntoChunks
+from DIRAC.Core.Security.ProxyInfo                      import getProxyInfo
+from DIRAC.Resources.Catalog.FileCatalogClientBase import FileCatalogClientBase
+
+from rucio.client import Client
+from rucio.common.exception import DataIdentifierNotFound, FileReplicaAlreadyExists, \
+                                   CannotAuthenticate, MissingClientParameter
+from rucio.common.utils import chunks, extract_scope
+sLog = gLogger.getSubLogger(__name__)
+
+
+def get_scope(lfn, scopes=[]):
+  """Helper function that extract the scope from the LFN
+  """
+  scope, _ = extract_scope(did=lfn, scopes=scopes)
```

Comment:
Even if `scopes` is never modified it's safer to do this:
```suggestion
def get_scope(lfn, scopes=None):
  """Helper function that extract the scope from the LFN
  """
  if scopes is None:
    scopes = []
  scope, _ = extract_scope(did=lfn, scopes=scopes)
```

Reply:
Done